mlos_bench/mlos_bench/services/types/remote_config_type.py

```diff
@@ -0,0 +1,56 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""
+Protocol interface for configuring cloud services.
+"""
+
+from typing import Any, Dict, Protocol, Tuple, TYPE_CHECKING, runtime_checkable
+
+if TYPE_CHECKING:
+    from mlos_bench.environments.status import Status
+
+
+@runtime_checkable
+class SupportsRemoteConfig(Protocol):
+    """
+    Protocol interface for configuring cloud services.
+    """
+
+    def configure(self, config: Dict[str, Any],
+                  params: Dict[str, Any]) -> Tuple["Status", dict]:
+        """
+        Update the parameters of a SaaS service in the cloud.
+
+        Parameters
+        ----------
+        config : Dict[str, Any]
+            Key/value pairs of configuration parameters (e.g., vmName).
+        params : Dict[str, Any]
```

Comment:
Agree - I've opened an issue #582 for this. I don't want to do it in this PR because this change will touch many other methods, e.g., in `AzureVMService`